File: backend/src/app/schemas/rag.py
"""Pydantic schemas for RAG search."""

from uuid import UUID

from pydantic import BaseModel, Field


class RagSearchRequest(BaseModel):
    customer_id: UUID
    query: str = Field(..., min_length=1, max_length=2000)
    ai_mode: bool = False
    top_k: int = Field(default=5, ge=1, le=20)


class ChunkResult(BaseModel):
    chunk_id: UUID
    attachment_id: UUID
    content: str
    page_number: int | None
    bbox: dict | None
    section_title: str | None
    score: float

    model_config = {"from_attributes": True}


class RagSearchResponse(BaseModel):
    chunks: list[ChunkResult]
    ai_answer: str | None = None

File: frontend/src/hooks/rag.ts
import { useMutation } from '@tanstack/react-query'
import { apiClient } from '@/lib/axios'

const BASE = '/api/v1/rag'

export interface RagSearchRequest {
  customer_id: string
  query: string
  ai_mode: boolean
  top_k?: number
}

export interface ChunkResult {
  chunk_id: string
  attachment_id: string
  content: string
  highlight: string | null
  page_number: number | null
  bbox: Record<string, unknown> | null
  section_title: string | null
  score: number
}

export interface RagSearchResponse {
  chunks: ChunkResult[]
  ai_answer: string | null
}

export function useRagSearch() {
  return useMutation({
    mutationFn: async (req: RagSearchRequest) => {
      const { data } = await apiClient.post<RagSearchResponse>(`${BASE}/search`, req)
      return data
    },
  })
}
